From e3f36db371d0a0ec8176d26bf4b5cd4f0c201287 Mon Sep 17 00:00:00 2001
From: Fabio Estevam <fabio.estevam@freescale.com>
Date: Mon, 4 May 2015 14:09:23 -0300
Subject: [PATCH] mx6cuboxi: Skip usb initialization when booting without HDMI

Starting USB initialization is useful for those who use Cuboxi/Hummingboard
with HDMI and USB keyboard.

However, when booting without a HDMI connection we can skip the usb
initialization, which makes the boot faster.

Signed-off-by: Jon Nettleton <jon.nettleton@gmail.com>
Signed-off-by: Fabio Estevam <fabio.estevam@freescale.com>
Tested-By: Vagrant Cascadian <vagrant@aikidev.net>
---
 include/configs/mx6cuboxi.h | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/include/configs/mx6cuboxi.h b/include/configs/mx6cuboxi.h
index 4e07f59..b6f9d4e 100644
--- a/include/configs/mx6cuboxi.h
+++ b/include/configs/mx6cuboxi.h
@@ -80,6 +80,7 @@
 #define CONFIG_VIDEO_LOGO
 #define CONFIG_VIDEO_BMP_LOGO
 #define CONFIG_IMX_HDMI
+#define CONFIG_CMD_HDMIDETECT
 #define CONFIG_IMX_VIDEO_SKIP
 #define CONFIG_CONSOLE_MUX
 
@@ -94,7 +95,17 @@
 #define CONFIG_USB_MAX_CONTROLLER_COUNT	2
 #define CONFIG_USB_KEYBOARD
 #define CONFIG_SYS_USB_EVENT_POLL
-#define CONFIG_PREBOOT			"usb start"
+#define CONFIG_PREBOOT \
+	"if hdmidet; then " \
+		"usb start; "		       \
+		"setenv stdin  serial,usbkbd; "\
+		"setenv stdout serial,vga; "   \
+		"setenv stderr serial,vga; "   \
+	"else " \
+		"setenv stdin  serial; " \
+		"setenv stdout serial; " \
+		"setenv stderr serial; " \
+	"fi;"
 
 #define CONFIG_SYS_NO_FLASH
 
@@ -117,9 +128,6 @@
 
 #define CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"stdin=serial,usbkbd\0" \
-	"stdout=serial,vga\0" \
-	"stderr=serial,vga\0" \
 	"script=boot.scr\0" \
 	"image=zImage\0" \
 	"fdtfile=undefined\0" \
-- 
2.1.4

